src/spyglass/position/v1/dlc_utils.py

```diff
@@ -714,30 +714,26 @@ def red_led_bisector_orientation(pos_df: pd.DataFrame, **params):
     LED2 = params.pop("led2", None)
     LED3 = params.pop("led3", None)
 
-    x_vec = pos_df[[LED1, LED2]].diff(axis=1).iloc[:, 0]
-    y_vec = pos_df[[LED1, LED2]].diff(axis=1).iloc[:, 1]
+    orient = np.full(len(pos_df), np.nan)  # Initialize with NaNs
+    x_vec = pos_df[LED1]["x"] - pos_df[LED2]["x"]
+    y_vec = pos_df[LED1]["y"] - pos_df[LED2]["y"]
 
-    y_is_zero = y_vec.eq(0)
-    perp_direction = pos_df[[LED3]].diff(axis=1)
+    # when y_vec is zero, 1&2 are equal. Compare to 3, determine if up or down
+    orient[y_vec.eq(0) & pos_df[LED3]["y"].gt(pos_df[LED1]["y"])] = np.pi / 2
+    orient[y_vec.eq(0) & pos_df[LED3]["y"].lt(pos_df[LED1]["y"])] = -np.pi / 2
 
-    # Handling the special case where y_vec is zero all Ys are the same
-    special_case = (
-        y_is_zero
-        & (pos_df[LED3]["y"] == pos_df[LED1]["y"])
-        & (pos_df[LED3]["y"] == pos_df[LED2]["y"])
-    )
-    if special_case.any():
+    # Handling error case where y_vec is zero and all Ys are the same
+    y_1, y_2, y_3 = pos_df[LED1]["y"], pos_df[LED2]["y"], pos_df[LED3]["y"]
+    if np.any(y_vec.eq(0) & y_1.eq(y_2) & y_1.eq(y_3)):
         raise Exception("Cannot determine head direction from bisector")
 
-    orientation = np.zeros(len(pos_df))
-    orientation[y_is_zero & perp_direction.iloc[:, 0].gt(0)] = np.pi / 2
-    orientation[y_is_zero & perp_direction.iloc[:, 0].lt(0)] = -np.pi / 2
-
-    orientation[~y_is_zero & ~x_vec.eq(0)] = np.arctan2(
-        y_vec[~y_is_zero], x_vec[~x_vec.eq(0)]
-    )
+    # General case where y_vec is not zero. Use arctan2 to determine orientation
+    length = np.sqrt(x_vec**2 + y_vec**2)
+    norm_x = (-y_vec / length)[~y_vec.eq(0)]
```

Comment:
Not a big deal but instead of computing equality to zero several times, it would probably be more efficient / clear to save it a variable. Also generally floating point number comparison should be computed using `np.isclose`

Are we assuming that if `y_vec` is zero in all the same places `x_vec` is?

Reply:
Sure, I can make those changes

I don't think that assumption is here - I think we're just handling the cases of `y_vec` being 0 differently - It there's enough info to determine up/down with X, do so. If all ys are 0 (because of bad data, misaligned leds, etc), raise the error.